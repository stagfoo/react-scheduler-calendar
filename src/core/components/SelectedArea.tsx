import React, { Component } from 'react';

interface SelectedAreaProps {
  schedulerData: any;
  left: number;
  width: number;
}

class SelectedArea extends Component<SelectedAreaProps> {
  constructor(props: SelectedAreaProps) {
    super(props);
  }

  render() {
    const { left, width, schedulerData } = this.props;
    const { config } = schedulerData;
    return (
      <div
        className="selected-area"
        style={{ left, width, top: 0, bottom: 0, backgroundColor: config.selectedAreaColor }}
      />
    );
  }
}

export default SelectedArea;
