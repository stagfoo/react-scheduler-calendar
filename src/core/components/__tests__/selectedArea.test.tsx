import { shallow, configure } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import SelectedArea from '../SelectedArea';

configure({ adapter: new Adapter() });

describe('selected area', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SelectedArea />);
    expect(wrapper).toMatchSnapshot();
  });
});
