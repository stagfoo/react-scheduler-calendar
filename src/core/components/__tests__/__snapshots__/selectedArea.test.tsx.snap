// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected area should render correctly 1`] = `
<div
  className="selected-area"
/>
`;
